fastcrawler/test_utils/test_engine.py

```diff
@@ -0,0 +1,108 @@
+from typing import Iterable
+from copy import copy
+import asyncio
+
+from .test_server import TestServer
+from fastcrawler.engine.contracts import Request, Response, Url
+
+
+class MockEngine:
+    """This class uses the TestServer instance to handle requests and return responses.
+
+    Attributes:
+        servers (TestServer | Iterable[TestServer]): The test server instances.
+        base_urls (str | Iterable[str]): The base urls of the requests.
+    """
+
+    def __init__(
+        self, servers: TestServer | Iterable[TestServer], base_urls: str | Iterable[str]
```

Comment:
servers: TestServer | Iterable[TestServer]
why is this Iterable of testserver, and testserver itself at the same time? wouldn't it be more meaningful if you just get the Iterable? and don't be too much flexible

Reply:
same goes for base_urls

Reply:
Shouldn't we be kind to the user? :)

ok it's fixed!

Reply:
> Shouldn't we be kind to the user? :)

No. I did same mistake in codebase. i will fix my own codes as well in another PR when i'm free.